src/OpenTelemetry/Trace/Span.cs

```diff
@@ -80,12 +96,14 @@ public sealed class Span : ISpan
             if (this.IsRecording)
             {
                 this.Activity.ActivityTraceFlags |= ActivityTraceFlags.Recorded;
-                
+
                 if (links != null)
                 {
+                    this.links = new EvictingQueue<Link>(this.tracerConfiguration.MaxNumberOfLinks);
```

Comment:
since we are now allowing links during span creation only, can we just have a list of links (`EvictingQueue` is not needed) and just take last `MaxNumberOfLinks` (if there are more than allowed?

Not a big deal from perf, but makes the intent of the code a bit cleaner

Reply:
That makes sense, sorry I missed that.

Reply:
Let's see if you'll let me pass with a `Reverse` 🙈